proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+5. **Support existing pipelines** that use KFP's standard artifact types (Dataset, Model, etc.) - pipelines work unchanged with the new filesystem backend
+6. **Match existing artifact persistence behavior** - artifacts persist indefinitely until explicitly deleted (no automatic cleanup)
+7. **Enable separate scaling of artifact serving** through an artifacts-only KFP instance with `--artifacts-only` flag
+8. **Ensure UI can properly handle artifact downloads** with the new `kfp-artifacts://` URI scheme
+
+### Non-Goals
+
+1. **Replace object storage** - filesystem storage is an additional option, not a replacement
+2. **Cross-namespace artifact sharing** - Initial implementation focuses on namespace isolation
+3. **Performance optimization for large-scale workloads** - filesystem storage may not match object storage performance for massive datasets or high concurrency scenarios
+4. **Storage migration tools** - Migrating from S3 to PVC/filesystem is not supported
+5. **Advanced storage features** - No versioning, replication, or geo-distribution initially
+
+## Proposal
+
+### Overview
+
+This KEP proposes adding a new artifact storage backend that uses filesystem storage (primarily Kubernetes `PersistentVolumeClaims`) instead of object storage. The implementation will:
+
+1. Create one PVC per namespace for artifact storage
+2. Use configurable access mode with sensible defaults (RWO)
+3. Organize artifacts in a filesystem hierarchy within the PVC
+4. Provide transparent access through the existing KFP artifact APIs with new `kfp-artifacts://` URI scheme
+5. Maintain compatibility with existing pipeline definitions that don't have hardcoded storage paths
+6. Support separate scaling of artifact serving through artifacts-only instances
+7. Update the UI to seamlessly handle artifact downloads from filesystem storage
+
+### User Stories
+
+#### Story 1: User Running Pipelines on Local Kubernetes
+
+As a user with KFP on kind/minikube/k3s, I want my pipeline artifacts to automatically use the local cluster's default `StorageClass` via the `kfp-artifacts://` scheme, so that I can develop pipelines offline without any storage configuration.
+
+**Acceptance Criteria:**
+
+- KFP works out-of-the-box with filesystem storage on local clusters
+- Artifacts are stored using the `kfp-artifacts://` URI scheme
+- No S3/GCS credentials required
+- Artifact viewing in UI works seamlessly
+
+#### Story 2: Operator Deploying KFP Without External Object Storage
+
+As an operator for a Kubeflow distribution, I want to deploy KFP with filesystem storage so that I don't need to productize and support a separate object storage system.
+
+**Acceptance Criteria:**
+
+- Single configuration option to enable filesystem storage
+- Artifact handling is part of KFP (no separate object storage component)
+- Storage automatically provisioned via PVCs
+- Backup/restore follows standard Kubernetes PVC procedures
```

Comment:
```suggestion
```

That is already covered by the default seaweedfs with zero effort. No storage configuration needed. seaweedfs just uses the default `StorageClass`. It already satisfies
- Artifact handling is part of KFP (no separate object storage component)
- Storage automatically provisioned via PVCs
- Backup/restore follows standard Kubernetes PVC procedures

So i recommend to remove this story